include/Core/UI/Setting/CoreSetting.php

```diff
@@ -212,4 +222,205 @@ public function render_hide_dashboard_widget_settings( $args ) {
+	 *
+	 * @param array $db_size_notification_data
+	 *
+	 * @return array $db_size_notification_data
+	 */
+	public function sanitize_db_size_notification( $db_size_notification_data ) {
+		$db_size_notification_data = array_filter( $db_size_notification_data,
+			array( $this, 'trim_array_to_allowed_keys' ),
+			ARRAY_FILTER_USE_KEY );
+
+		foreach ( $db_size_notification_data as $setting => $value ) {
+			if ( $setting === 'notify' ) {
+				$db_size_notification_data[ $setting ] = \filter_var( $value, FILTER_VALIDATE_BOOLEAN );
+			} elseif ( $setting === 'admin_email' ) {
+				$db_size_notification_data[ $setting ] = \sanitize_email( $value );
+			} elseif ( $setting === 'logs_threshold' ) {
+				$db_size_notification_data[ $setting ] = intval( \sanitize_text_field( $value ) );
+			}
+		}
+
+		// wp_parse_args won't merge nested array keys. So add the key here if it is not set.
+		if ( ! array_key_exists( 'notify', $db_size_notification_data ) ) {
+			$db_size_notification_data['notify'] = false;
+		}
+		if ( ! array_key_exists( 'log_threshold_met', $db_size_notification_data ) ) {
+			$db_size_notification_data['log_threshold_met'] = false;
+		}
+
+		return $db_size_notification_data;
+	}
+
+	/**
+	 * Triggers the Cron to notify admin via email.
+	 *
+	 * Email is triggered only when the threshold setting is met.
+	 *
+	 * @since 2.3.0
+	 */
+	public function verify_email_log_threshold() {
+		$cron_hook = 'el_trigger_notify_email_when_log_threshold_met';
+		if ( ! wp_next_scheduled( $cron_hook ) ) {
+			wp_schedule_event( time(), 'hourly', $cron_hook );
+		}
+	}
+
+	/**
+	 * Send the Threshold notification email to the admin.
+	 *
+	 * @since 2.3.0
+	 */
+	public function trigger_threshold_met_notification_email() {
+		$email_log  = email_log();
+		$logs_count = intval( $email_log->table_manager->get_logs_count() );
+
+		$setting_data = $this->get_value();
+
+		// Return early.
+		if ( ! array_key_exists( 'db_size_notification', $setting_data ) ) {
+			return;
+		}
+
+		$db_size_notification_key  = 'db_size_notification';
+		$db_size_notification_data = $setting_data[ $db_size_notification_key ];
+
+		// Return early.
+		if ( ! array_key_exists( 'logs_threshold', $db_size_notification_data ) ||
+		     ! array_key_exists( 'admin_email', $db_size_notification_data ) ||
+		     ! array_key_exists( 'notify', $db_size_notification_data ) ) {
+			return;
+		}
+
+		$is_notification_enabled = $db_size_notification_data['notify'];
+		$admin_email             = $db_size_notification_data['admin_email'];
+		$logs_threshold          = intval( $db_size_notification_data['logs_threshold'] );
+		$logs_threshold_met      = $db_size_notification_data['log_threshold_met'];
+
+		// Ideally threshold cannot be 0. Also, skip sending email if it is already sent.
+		if ( $logs_threshold === 0 || $logs_threshold_met === true ) {
+			return;
+		}
+
+		if ( $is_notification_enabled && is_email( $admin_email ) && ( $logs_count >= $logs_threshold ) ) {
```

Comment:
Split `is_notification_enabled` check from this condition and do this before checking `logs_threshold` above.

Reply:
Fixed.